hedera-mirror-rest/service/baseService.js

```diff
@@ -0,0 +1,46 @@
+/*-
+ * ‌
+ * Hedera Mirror Node
+ * ​
+ * Copyright (C) 2019 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+'use strict';
+
+const _ = require('lodash');
+
+/**
+ * Base service class that other services should inherit from for their retrieval business logic
+ */
+class BaseService {
+  getOrderByQuery(column, order) {
+    return `order by ${column} ${order}`;
+  }
+
+  getLimitQuery(limitParamCount) {
+    return `limit $${limitParamCount}`;
+  }
+
+  async getRows(query, params, functionName = '') {
+    logger.trace(`${functionName} query: ${query}, params: ${params}`);
+    return pool.queryQuietly(query, params).then((result) => {
+      logger.trace(`${functionName} ${result.rows.length} entries`);
+      return result.rows;
+    });
```

Comment:
better not mix async / await and thenable:

```suggestion
   const {rows} = await pool.queryQuietly(query, params);
   logger.trace(`${functionName} ${rows.length} entries`);
   return result.rows;
```

Reply:
I believe I had this initially as you stated and addressed feedback to make it thenable.
Can revert

Reply:
Reverted

Reply:
yup. it still returns a promise. from a caller's pointer of view, there's no difference. however `await` makes the code more readable since it just looks like sync code.